### PR TITLE
Signup and login modal

### PR DESCRIPTION
**Pull request by davideastmond**
- Login and signup
- Validate the fields using rules. 
- The header bar opens the modals

image

image

image

### CODE REVIEW

**Comment by LifeAsDev** [comment]
that is a lot of code, we can talk it on the next meet

File: travel-event-host-app/src/components/auth-dialog/AuthDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+ */
+
+interface AuthDialogProps {
+  open: boolean;
+  onDialogClose: () => void;
+  authDialogType: 'login' | 'signup';
+}
+
+export default function AuthDialog(props: AuthDialogProps) {
+  const [errors, setErrors] = useState<Record<string, string[]>>({});
```
Comment (davideastmond) [inline travel-event-host-app/src/components/auth-dialog/AuthDialog.tsx:42]: @LifeAsDev In the meeting you mentioned the case where the user tries to register with an e-mail that is already in use We should be able to add the back-end error response to the `errors` state object.

Comment (LifeAsDev) [reply travel-event-host-app/src/components/auth-dialog/AuthDialog.tsx:42]: yeah, what message I have to return from the backend?

Comment (davideastmond) [reply travel-event-host-app/src/components/auth-dialog/AuthDialog.tsx:42]: You could follow a similar structure and return an Record<string, string[]>. In the backend you have the message property return
{ email: ['Email already in use'] }
```

**Review by LifeAsDev — APPROVED** [review]

**Pull request merged**